### DISCUSSION TITLE
[Tex:Fraction&Super/Subscript]分数输入与上下标

### DISCUSSION THREAD

**Discussion by suucss** [3b1b/manim#1535] [Q&A] [answered]
我想输入一个含有分数线的式子，分子和分母均含有下标，如下图
![1](https://user-images.githubusercontent.com/84609589/121797190-42cbc580-cc51-11eb-8a41-bf90c822f12f.png)
然鹅出现了latex error converting to dvi.
经过多次试验，我发现只有在分子和分母均不含上下标时没有报错，分子或分母只要其中一个含有上下标就会报错，如：
正确代码示例：frac=Tex("$a=\\frac{b}{2}$")【分数不含上下标】
报错代码示例：frac2=Tex("$F=G\\frac{m_{1}m_{2}}{r^{2}}$")【分数含有上下标】
我猜测是由于多重{}符号导致manim将识别结束提前（这不太可能），我想知道如何解决分子/分母携带上下标时报错的问题
万分感谢！

**Comment by TonyCrane** [comment, marked as answer]
你使用的是Tex，所以你不用在传入字符串里加入$$
这样使用就没有问题了：
```python
frac = Tex("F=G\\frac{m_{1}m_{2}}{r^{2}}")
```
TexText是直接的LaTeX环境，编写公式需要$$
Tex是LaTeX的align环境，编写公式不需要$$，但写文字需要\text{}
所以这个和下面使用TexText的效果是一样的：
```python
frac = TexText("$F=G\\dfrac{m_{1}m_{2}}{r^{2}}$") # 使用\dfrac启用display style来达到一样的大小
```
![image](https://user-images.githubusercontent.com/44120331/121797952-08185c00-cc56-11eb-97f0-ec3ea7c98b53.png)

**Reply by suucss** [reply]
十分感谢您的回答，但是，使用Tex的那段代码，发生报错如下（看到报错真是太难受了）
![QQ截图20210613145854](https://user-images.githubusercontent.com/84609589/121798299-09e31f00-cc58-11eb-8e20-7bf1153d8f68.png)
在TexText中，编译器PyCharm提示没有TexText
![QQ截图20210613150006](https://user-images.githubusercontent.com/84609589/121798366-6b0af280-cc58-11eb-89b7-ae840695aaa2.png)
我使用的是https://github.com/3b1b/manim 版本，时间是2021-04-28
![QQ截图20210613150036](https://user-images.githubusercontent.com/84609589/121798397-9db4eb00-cc58-11eb-8c44-28dbc5b4f650.png)

> frac = Tex("F=G\\frac{m_{1}m_{2}}{r^{2}}")
   self.play(FadeInFrom(frac, LEFT))
   self.wait(1)

**Reply by TonyCrane** [reply]
但是我根据你报错的样式推断你安装的应该是community版本。3b1b/manim不会有这样的输出。
而且TexText在3b1b/manim中是有的，而community版本的manim没有TexText

**Reply by suucss** [reply]
![QQ截图20210613153602](https://user-images.githubusercontent.com/84609589/121799157-fbe3cd00-cc5c-11eb-8539-aec296651f6f.png)
应该是的

**Reply by suucss** [reply]
大佬在上，接受萌新一拜

**Comment by suucss** [comment]
[ResultsFeedback]成功！
![1](https://user-images.githubusercontent.com/84609589/121809213-84c72c80-cc8e-11eb-9acb-b62b89069d7b.png)
![2](https://user-images.githubusercontent.com/84609589/121809218-8abd0d80-cc8e-11eb-90f2-e303685dabb6.png)
反思：

1. 使用最新最优的代码格式，如使用MathTex("")而非Tex("")，后者不工作；
2. 使用标准的语法格式，句法空格等，m_{1}v_{1} 不工作，m_{1}(空格)v_{1} 能够识别；（好奇XG是不是在编译时，编译器自动更正格式？）
3. 《论大佬的重要性》：-D
